precovery/frame_db.py

```diff
@@ -174,6 +176,11 @@ def get_frames(
     ) -> Iterator[HealpixFrame]:
         """
         Yield all the frames which are for given obscode, MJD, healpix.
+
+        MJDs are checked to within +- 1e-7 days or 8.64 ms. Any frames that
+        are within 8.64 ms of the given mjd will be returned. This does not garauntee
+        that they will represent the desired exposure time and may lead to multiple
+        matches computed at the wrong observation time.
```

Comment:
Oof, this is annoying, isn't it?

This is just a problem with floating point timestamps, I suppose. Perhaps there is a way we can log at the WARN level if we get multiple hits?

Reply:
I'd suggest we reduce the 1e-7 further to 1e-12 or 0.0864 microseconds (what is the smallest comparison we could go to numerically?). We should hopefully never (🤞) encounter two exposures within 0.0864 microseconds from the same observatory. 

Checking for MJD uniqueness would require (I think) looping over the rows first before yielding them back and tracking the unique number of MJDs. It's not fully obvious how much more expensive that would be but it is something that is totally doable. Something as simple as this could work? 
```
rows = self.dbconn.execute(select_stmt)

mjds = set()
for r in rows:
    mjds.add(r[3])

if len(mjds) > 1:
   logger.warn(f"Query returned non-unique MJDs for mjd: {mjd}, healpix: {int(healpixel)}, obscode: {obscode}.")
 
for r in rows:
    yield HealpixFrame(*r)
```

Reply:
This is really fraught with peril. Floating point inaccuracies are really hard to predict. See https://www.sqlite.org/floatingpoint.html - a serious applied mathematician could probably put error bounds on this but that's way beyond my abilities.

Your loop is valid and fast, and the warning is probably sufficient. I think that's very preferable over shrinking down to less than a microsecond.

The real Right Thing here is to stop using floating point values. We should store times as integers if we want to do exact matching. That could be done by using something like UTC milliseconds, which should map cleanly from Julian Dates. Since sqlite uses 64-bit integers, this should be valid for essentially forever. Another option would be to multiply our MJDs by a constant integer factor (say, 1e9) and round to the nearest integer.

Then, we could actually do exact matches and thing should work correctly.

Reply:
I've added a logging statement at the WARN level in [0799cd1](https://github.com/B612-Asteroid-Institute/precovery/pull/8/commits/0799cd128f1e8da670f7c003cb35294af3f8dcd2) and kept the time range at +- 1e-7 days.